types/address_test.go

```diff
@@ -339,34 +339,29 @@ func (s *addressTestSuite) TestAddressInterface() {
 }
 
 func (s *addressTestSuite) TestVerifyAddressFormat() {
-	addr0 := make([]byte, 0)
-	addr5 := make([]byte, 5)
-	addr20 := make([]byte, 20)
-	addr32 := make([]byte, 32)
-	addr256 := make([]byte, 256)
-
-	err := types.VerifyAddressFormat(addr0)
-	s.Require().EqualError(err, "addresses cannot be empty: unknown address")
-	err = types.VerifyAddressFormat(addr5)
-	s.Require().NoError(err)
-	err = types.VerifyAddressFormat(addr20)
-	s.Require().NoError(err)
-	err = types.VerifyAddressFormat(addr32)
-	s.Require().NoError(err)
-	err = types.VerifyAddressFormat(addr256)
-	s.Require().EqualError(err, "address max length is 255, got 256: unknown address")
+	addr1, err := address.NewHexCodec().StringToBytes("0x000000000000000000000000000000000000dead")
+	require.NoError(s.T(), err)
+
+	err = types.VerifyAddressFormat(addr1)
```

Comment:
Inside VerifyAddressFormat I see that we don't check length of bytes, I think its good idea to add check that they should be exactly 20 bytes. Public key address function will return exactly 20 bytes always - https://github.com/0xPolygon/cosmos-sdk/blob/mardizzone/POS-2455/crypto/keys/secp256k1/secp256k1.go#L219-L224 ofcourse during some transformations maybe zeroes which were as suffix or prefix could be lost, so maybe we just check if its up to 20 bytes.

Reply:
Yeah, I skipped that just because the following code encodes up to the `len(s)` but it makes sense to have. I'd actually go for `!= 20` as in `VerifyAddressFormat` we only accepts addresses bytes, hence the hard limit on length should always be respected.
```
	if !common.IsHexAddress(common.Bytes2Hex(bz)) {
		return errorsmod.Wrapf(sdkerrors.ErrUnknownAddress, "invalid address")
	}
```
Addressed.